src/storage/src/client.rs

```diff
@@ -554,6 +590,34 @@ mod tests {
             .expect_err("malformed bucket string should error");
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_read_object_query_params() -> Result {
+        let client = Storage::builder()
+            .with_credentials(auth::credentials::testing::test_credentials())
+            .build()
+            .await?;
+
+        let read_object_builder = client
+            .read_object()
+            .set_bucket("projects/_/buckets/bucket")
+            .set_object("object")
+            .set_generation(5)
+            .set_if_generation_match(10)
+            .set_if_generation_not_match(20)
+            .set_if_metageneration_match(30)
+            .set_if_metageneration_not_match(40)
+            .http_request_builder()
+            .await?
+            .build()?;
+
+        assert_eq!(read_object_builder.method(), reqwest::Method::GET);
+        assert_eq!(
+            read_object_builder.url().as_str(),
+            "https://storage.googleapis.com/storage/v1/b/bucket/o/object?alt=media&generation=5&ifGenerationMatch=10&ifGenerationNotMatch=20&ifMetagenerationMatch=30&ifMetagenerationNotMatch=40"
+        );
```

Comment:
nit: I do not love that this depends on the order of the arguments and that it is a long string.  I think you can rewrite this as:

```suggestion
            concat!("https://storage.googleapis.com/storage/v1/b/bucket/o/object?",
          "alt=media&",
          "generation=5&",
          "ifGenerationMatch=10&",
          "ifGenerationNotMatch=20&",
          "ifMetagenerationMatch=30&",
          "ifMetagenerationNotMatch=40")
        );
```

In the future, consider something like: https://docs.rs/reqwest/latest/reqwest/struct.Url.html#method.query_pairs to compare the query parameters as (sorted) vectors or something.